lib/SPIRVProducerPass.cpp

```diff
@@ -4324,10 +4334,26 @@ SPIRVID SPIRVProducerPassImpl::GenerateInstructionFromCall(CallInst *Call) {
 
 void SPIRVProducerPassImpl::GenerateInstruction(Instruction &I) {
   ValueMapType &VMap = getValueMap();
+  DIFileMap &DbgDIFileMap = getDebugDIFileMap();
   LLVMContext &Context = module->getContext();
 
   SPIRVID RID;
 
+  if (auto md = I.getMetadata("dbg")) {
```

Comment:
Ideally we would only generate one OpLine instruction for a chain of instructions that share a location. Can we save the debug info (reset every basic block) and check if it differs?

Separately, I think this needs an else that if we want debug info and there is no info (and we're in a block) we should insert an OpNoLine instruction to prevent the previous OpLine from affecting this instruction.